WooCommerce/Classes/Extensions/NumberFormatter+Localized.swift

```diff
@@ -29,4 +29,35 @@ extension NumberFormatter {
         formatter.roundingMode = .halfUp
         return formatter.string(from: number)
     }
+
+    /// Converts given number in string format, from the specified source locale to target locale.
+    ///
+    /// This method does not accept numbers with grouping separator. (No thousands separator)
+    ///
+    /// - Parameters:
+    ///     - using: The string to be localized.
+    ///     - from: The current `Locale` of the input string.
+    ///     - to: The `Locale` to be used for localizing the input string.
+    ///
+    /// - Returns: The input string localized to the target locale. Returns `nil` if the localization is unsucessful.
+    ///
+    static func localizedString(using string: String,
+                                from sourceLocale: Locale,
+                                to targetLocale: Locale) -> String? {
```

Comment:
non-blocking 💭 : this feels a bit specific to product shipping values, like the assumption of no grouping separator. I was wondering if this function could be just for formatting the product shipping values instead of a generic function, so that we can encapsulate the logic for other assumptions like the API using US locale in `DefaultProductFormTableViewModel` below:

```swift
func localizedNumber(_ string: String) -> String? {
            // API uses US locale for weight and shipping dimensions
            let usLocale = Locale(identifier: "en_US")
            return NumberFormatter.localizedString(using: string, from: usLocale, to: locale)
}
```

Like this function could be `static func localizedShippingValueString(using string: String, to targetLocale: Locale) -> String?` and we can set the US locale in L48 below. Then anywhere we want to show localized shipping values, like in both product form and `ProductShippingSettingsViewModel`, we can use this helper function to do so. WDYT?

Reply:
I saw that later in the PR a reverse conversion is needed in `func formattedForAPI(fromLocale: Locale)`. Maybe the current `static func localizedString` could be kept this way as private, and having two public functions like `static func localizedShippingValueString` and `static func unlocalizedShippingValueString` for conversions in two ways?

Keeping these helper function(s) in a separate struct/class could also be nice to not add too many functionalities to `NumberFormatter` 😄

Reply:
> Keeping these helper function(s) in a separate struct/class could also be nice to not add too many functionalities to NumberFormatter

💯 

I was trying to think of ways to make this better, but I didn't think about totally moving it outside of `NumberFormatter`. This is a great suggestion. Thank you! 🙇